src/frameworks/next/utils.ts

```diff
@@ -489,3 +492,52 @@ export async function whichNextConfigFile(dir: string): Promise<NextConfigFileNa
 
   return null;
 }
+
+/**
+ * Helper function to find the path of esbuild using `npm which`
+ */
+export function findEsbuildPath(): string | null {
+  try {
+    const esbuildBinPath = execSync("npx which esbuild", { encoding: "utf8" }).trim();
+    const globalVersion = getGlobalEsbuildVersion();
+    if (globalVersion && !satisfies(globalVersion, ESBUILD_VERSION)) {
+      console.warn(
+        `Warning: Global esbuild version (${globalVersion}) does not match the required version (${ESBUILD_VERSION}).`,
+      );
+    }
+    return resolve(dirname(esbuildBinPath), "../esbuild");
```

Comment:
does `../esbuild` resolves correctly on all platforms? I'm under the impression we need to use `join` here

Reply:
@leoortizz `resolve` will indeed resolve correctly on all platforms as it normalizes the path and converts it to an absolute path.